packages/experiments/src/components/UnifiedPicker/UnifiedPicker.tsx

```diff
@@ -94,18 +86,56 @@ export const UnifiedPicker = <T extends {}>(props: IUnifiedPickerProps<T>): JSX.
     return dragEnterClass;
   };
 
+  const _dropItemsAt = (insertIndex: number, s: string): void => {
+    if (props.selectedItemsListProps.getDeserializedItems) {
+      const newItems = props.selectedItemsListProps.getDeserializedItems(s);
+      let indicesToRemove: number[] = [];
+      // If we are moving items within the same picker, remove them from their old places as well
+      if (draggedIndex > -1) {
+        indicesToRemove = focusedItemIndices.includes(draggedIndex) ? [...focusedItemIndices] : [draggedIndex];
+      }
+      if (props.selectedItemsListProps.dropItemsAt) {
+        props.selectedItemsListProps.dropItemsAt(insertIndex, newItems, indicesToRemove);
+      }
+      dropItemsAt(insertIndex, newItems, indicesToRemove);
+    }
+  };
+
   const _onDrop = (item?: any, event?: DragEvent): void => {
-    if (draggedItem) {
-      _insertBeforeItem(item);
+    if (event?.dataTransfer) {
+      event.preventDefault();
+      const data = event.dataTransfer.items;
+      for (let i = 0; i < data.length; i++) {
+        if (data[i].kind === 'string' && data[i].type === props.customClipboardType) {
```

Comment:
With these changes, the UPP will only do drag drop if the serialized/deserialized functions are defined. Should I add secondary logic to say if there isn't data transfer data, but is a dragged item index, move that?

I think probably yes thinking about it